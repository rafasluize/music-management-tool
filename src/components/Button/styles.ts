import styled from 'styled-components';

const Button = styled.button<{ variant: 'neutral' | 'brand' }>`

  
    background:  ${({ theme, variant }) =>
      variant === 'brand'
        ? `
    linear-gradient(
        90deg,
        ${theme.color.background.brand.default} 0%,
        ${theme.color.background.secondary.default} 91%
      );
    `
        : `${theme.color.background.neutral.inverted};`}

    
    
    color: white;
    padding: 5px 15px;
    border-radius: 5px;
    outline: 0;
    border: 0; 
    margin: 10px 0px;
    cursor: pointer;
    box-shadow: 0px 2px 2px lightgray;
    transition: ease background-color 250ms;
    &:hover {
        ${({ theme, variant }) =>
          variant === 'brand'
            ? `
            background:  linear-gradient(
            90deg,
            ${theme.color.background.secondary.default} 0%,
            ${theme.color.background.brand.default} 91%
          );
        `
            : `background:  ${theme.color.background.neutral.strong};
            color: ${theme.color.text.neutral.strong}`}
          
    }
    &:disabled {
      cursor: default;
      opacity: 0.7;
    }
  `;

export { Button };
