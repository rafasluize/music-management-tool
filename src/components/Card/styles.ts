import styled from 'styled-components';

const Wrapper = styled.section`
  background-color: ${({ theme }) => theme.color.background.neutral.default};
  border-radius: ${({ theme }) => theme.border.radius.medium};
  box-shadow: ${({ theme }) => theme.elevation.ton5};
`;

const Header = styled.div`
  border-bottom: ${({ theme }) => theme.border.width.thinner} solid
    ${({ theme }) => theme.color.background.neutral.strong};
  padding: ${({ theme }) => theme.spacing.km4};
  display: flex;
  align-items: center;
  justify-content: space-between;

  color: ${({ theme }) => theme.color.text.secondary.default};
  font-weight: 700;
  font-size: ${({ theme }) => theme.font.size.m6};

  div {
    font-weight: 700;
    font-size: ${({ theme }) => theme.font.size.m2};
    color: ${({ theme }) => theme.color.text.neutral.strong};
  }
`;

const Content = styled.div`
  padding: ${({ theme }) => theme.spacing.km4};
`;

export { Wrapper, Header, Content };
