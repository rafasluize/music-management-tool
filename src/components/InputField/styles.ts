import { ErrorMessage } from 'formik';
import styled from 'styled-components';

const Wrapper = styled.div`
margin-bottom: ${({ theme }) => theme.spacing.km4};

   > label {
    display: block;
  font-size: ${({ theme }) => theme.font.size.m3};
  margin-bottom: ${({ theme }) => theme.spacing.km1};



   }

   > input {
    width: 100%;
  height: 30px;

  &:focus {
    outline: ${({ theme }) => theme.border.width.thinner} solid ${({ theme }) => theme.color.text.brand.default};

  }


s

   }
`;

const ErrorMessageCustom = styled(ErrorMessage)`
  color: ${({ theme }) => theme.color.text.danger.default};
  font-size: ${({ theme }) => theme.font.size.m3};
`;

export { Wrapper, ErrorMessageCustom };
