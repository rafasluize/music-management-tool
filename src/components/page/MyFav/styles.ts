import styled from 'styled-components';

const Wrapper = styled.section`
  width: 100%;
  margin-top: ${({ theme }) => theme.spacing.km4};
`;

const Thead = styled.div`
  display: flex;
  position: sticky;
  top: 0;
  background-color: ${({ theme }) => theme.color.background.neutral.weak};
  div {
    width: 30%;
    padding: ${({ theme }) => theme.spacing.km3};
    display: flex;
    justify-content: space-between;
    cursor: pointer;

    &:not(:last-child) {
      border-right: ${({ theme }) => theme.border.width.thinner} solid ${({ theme }) => theme.color.text.neutral.weak};
    }

    &:first-child {
      width: 40%;
    }
  }
`;

const Content = styled.div`
  scrollbar-width: thin;
  max-height: 550px;
  display: block;
  overflow-y: auto;
`;

const Item = styled.div`
  display: flex;

  div {
    font-size: ${({ theme }) => theme.font.size.m2};
    padding: ${({ theme }) => theme.spacing.km4};
    color: ${({ theme }) => theme.color.text.neutral.strong};

    width: 30%;

    &:first-child {
      width: 40%;
    }
  }

  &:hover {
    background-color: ${({ theme }) => theme.color.background.secondary.weak};

    div {
      color: ${({ theme }) => theme.color.text.secondary.default};
    }
  }
`;

const Buttons = styled.div`
  display: flex;
  gap: 16px;
`;

export { Wrapper, Item, Thead, Content, Buttons };
