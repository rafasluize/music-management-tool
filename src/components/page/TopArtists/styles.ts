import styled from 'styled-components';

const Wrapper = styled.div`
  background-color: ${({ theme }) => theme.color.background.neutral.strong};
  border-radius: ${({ theme }) => theme.border.radius.medium};
  padding: ${({ theme }) => theme.spacing.km4};
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.spacing.km2};
`;

const Item = styled.div`
  background-color: ${({ theme }) => theme.color.background.neutral.default};
  border-radius: ${({ theme }) => theme.border.radius.small};
  padding: ${({ theme }) => theme.spacing.km3};
  display: flex;
  justify-content: space-between;

  span {
    color: ${({ theme }) => theme.color.text.secondary.default};
    font-weight: 700;
    font-size: ${({ theme }) => theme.font.size.m4};
  }

  &:hover {
    outline: ${({ theme }) => theme.border.width.thinner} solid ${({ theme }) => theme.color.text.brand.default};
  }
`;

export { Wrapper, Item };
