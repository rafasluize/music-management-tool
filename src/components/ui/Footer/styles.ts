import styled from 'styled-components';

const Wrapper = styled.header`
  padding: ${({ theme }) => theme.spacing.km4};

  text-align: center;
  background-color: ${({ theme }) => theme.color.background.neutral.strong};
`;

export { Wrapper };
