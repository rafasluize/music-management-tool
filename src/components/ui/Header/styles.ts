import styled from 'styled-components';

const Wrapper = styled.header`
  padding: ${({ theme }) => theme.spacing.km5} ${({ theme }) => theme.spacing.km8};
  display: flex;
  align-items: center;
  background-color: ${({ theme }) => theme.color.background.brand.default};
  background: linear-gradient(
    90deg,
    ${({ theme }) => theme.color.background.brand.default} 0%,
    ${({ theme }) => theme.color.background.secondary.default} 91%
  );

  h1 {
    color: ${({ theme }) => theme.color.text.neutral.inverted};
    font-size: ${({ theme }) => theme.font.size.m6};
  }
`;

const User = styled.div<{ letter: string }>`
  margin-left: auto;
  color: ${({ theme }) => theme.color.text.neutral.inverted};
  font-size: ${({ theme }) => theme.font.size.m4};

  &::after {
    content: '${({ letter }) => letter}';
    background-color: ${({ theme }) => theme.color.background.neutral.default};
    color: ${({ theme }) => theme.color.text.neutral.strong};
    font-weight: ${({ theme }) => theme.font.weight.bold};

    padding: ${({ theme }) => theme.spacing.km3};
    margin-left: ${({ theme }) => theme.spacing.km3};
    border-radius: ${({ theme }) => theme.border.radius.circle};
    width: 40px;
    height: 40px;
  }
`;

export { Wrapper, User };
