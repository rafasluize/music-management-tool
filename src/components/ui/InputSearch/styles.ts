import { ErrorMessage } from 'formik';
import styled from 'styled-components';

const Wrapper = styled.div`
  margin-bottom: ${({ theme }) => theme.spacing.km4};

  > div {
    outline: ${({ theme }) => theme.border.width.thinner} solid ${({ theme }) => theme.color.text.neutral.weak};
    display: flex;
    align-items: center;

    > input {
      height: 30px;
      border: none;
      width: calc(100% - 16px);

      &:focus {
        outline: none;
      }
    }
  }

  > label {
    display: block;
    font-size: ${({ theme }) => theme.font.size.m3};
    margin-bottom: ${({ theme }) => theme.spacing.km1};
  }
`;

const ErrorMessageCustom = styled(ErrorMessage)`
  color: ${({ theme }) => theme.color.text.danger.default};
  font-size: ${({ theme }) => theme.font.size.m3};
`;

export { Wrapper, ErrorMessageCustom };
