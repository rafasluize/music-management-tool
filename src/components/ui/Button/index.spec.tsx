import React from 'react';
import { render } from 'test-utils/render';
import { Button } from './styles';

describe('button', () => {
  it('neutral', () => {
    const { container } = render(<Button variant="neutral">text</Button>);

    // eslint-disable-next-line testing-library/no-node-access
    const wrapper = container.firstElementChild;

    expect(wrapper).toMatchSnapshot();
  });

  it('brand', () => {
    const { container } = render(<Button variant="brand">text</Button>);

    // eslint-disable-next-line testing-library/no-node-access
    const wrapper = container.firstElementChild;

    expect(wrapper).toMatchSnapshot();
  });
});
