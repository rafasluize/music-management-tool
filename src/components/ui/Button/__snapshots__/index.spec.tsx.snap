// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button brand 1`] = `
.c0 {
  background: linear-gradient(         90deg,         #dd2a62 0%,         #6c2c83 91%       );
  color: white;
  padding: 5px 15px;
  border-radius: 5px;
  outline: 0;
  border: 0;
  cursor: pointer;
  box-shadow: 0px 2px 2px lightgray;
  transition: ease background-color 250ms;
}

.c0:hover {
  background: linear-gradient(             90deg,             #6c2c83 0%,             #dd2a62 91%           );
}

.c0:disabled {
  cursor: default;
  opacity: 0.7;
}

<button
  class="c0"
  variant="brand"
>
  text
</button>
`;

exports[`button neutral 1`] = `
.c0 {
  background: #111111;
  color: white;
  padding: 5px 15px;
  border-radius: 5px;
  outline: 0;
  border: 0;
  cursor: pointer;
  box-shadow: 0px 2px 2px lightgray;
  transition: ease background-color 250ms;
}

.c0:hover {
  background: #dddfe4;
  color: #111111;
}

.c0:disabled {
  cursor: default;
  opacity: 0.7;
}

<button
  class="c0"
  variant="neutral"
>
  text
</button>
`;
